test/org/sagebionetworks/bridge/play/controllers/SharedModuleMetadataControllerTest.java

```diff
@@ -87,33 +88,156 @@ public void create() throws Exception {
+        Result result = controller.deleteMetadataByIdAndVersion(MODULE_ID, 3, "true");
+        TestUtils.assertResult(result, 200);
+
+        verify(mockSvc).deleteMetadataByIdAndVersion(MODULE_ID, 3);
+        verify(controller).getAuthenticatedSession(Roles.DEVELOPER, Roles.ADMIN);
+    }
+    
+    @Test
+    public void deleteMetadataByIdAndVersionAdminPhysical() throws Exception {
+        mockSession.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(Roles.ADMIN)).build());
+        doReturn(mockSession).when(controller).getAuthenticatedSession(Roles.DEVELOPER, Roles.ADMIN);
+        
+        // setup, execute, and validate
+        Result result = controller.deleteMetadataByIdAndVersion(MODULE_ID, 3, "true");
         TestUtils.assertResult(result, 200);
 
         // verify backend
-        verify(mockSvc).deleteMetadataByIdAndVersion(MODULE_ID, MODULE_VERSION);
+        verify(mockSvc).deleteMetadataByIdAndVersionPermanently(MODULE_ID, 3);
+        verify(controller).getAuthenticatedSession(Roles.DEVELOPER, Roles.ADMIN);
+    }
+    
+    @Test
+    public void deleteByIdAllVersionsPermanently() throws Exception {
+        mockSession.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(Roles.ADMIN)).build());
+        doReturn(mockSession).when(controller).getAuthenticatedSession(Roles.DEVELOPER, Roles.ADMIN);
+        
+        // setup, execute, and validate
+        Result result = controller.deleteMetadataByIdAllVersions(MODULE_ID, "true");
+        TestUtils.assertResult(result, 200);
+
+        // verify backend
+        verify(mockSvc).deleteMetadataByIdAllVersionsPermanently(MODULE_ID);
 
         // validate permissions
-        verify(controller).getAuthenticatedSession(Roles.DEVELOPER);
+        verify(controller).getAuthenticatedSession(Roles.DEVELOPER, Roles.ADMIN);
+    }
+    
+    @Test
+    public void deleteByIdAndVersionPermanently() throws Exception {
+        mockSession.setParticipant(new StudyParticipant.Builder().withRoles(ImmutableSet.of(Roles.ADMIN)).build());
+        doReturn(mockSession).when(controller).getAuthenticatedSession(Roles.DEVELOPER, Roles.ADMIN);
+        
+        // setup, execute, and validate
+        Result result = controller.deleteMetadataByIdAndVersion(MODULE_ID, MODULE_VERSION, "true");
+        TestUtils.assertResult(result, 200);
+
+        // verify backend
+        verify(mockSvc).deleteMetadataByIdAndVersionPermanently(MODULE_ID, MODULE_VERSION);
+
+        // validate permissions
+        verify(controller).getAuthenticatedSession(Roles.DEVELOPER, Roles.ADMIN);
+    }
+
+    @Test
+    public void deleteByIdAndVersionLogically() throws Exception {
     }
 
+    @Test
+    public void deleteByIdAndVersionDefaultsToLogicalDelete() throws Exception {
+    }
+    
+    @Test
+    public void deleteByIdAndVersionByDeveloperDefaultsToLogicalDelete() throws Exception {
+    }
```

Comment:
Sorry, these are redundant. Removing. The tests for the logical delete are there, they just don't say "logical" in the test name... the existing delete tests because the logical delete tests.